tastytaps/taps/api.py

```diff
@@ -0,0 +1,30 @@
+from rest_framework import generics
+from rest_framework.permissions import IsAuthenticated
+
+from .models import Brewery, Taps
+from .serializers import BrewerySerializer, TapsListSerializer, TapsSerializer
+
+
+class BreweryListCreate(generics.ListCreateAPIView):
+    permission_classes = [IsAuthenticated]
+    queryset = Brewery.objects.all()
+    serializer_class = BrewerySerializer
+
+
+class TapsMixin(object):
+    permission_classes = [IsAuthenticated]
+    queryset = Taps.objects.all()
+
+
+class TapsCreate(TapsMixin, generics.CreateAPIView):
+    serializer_class = TapsSerializer
+
+
+class TapsDetail(TapsMixin, generics.RetrieveAPIView):
+    """Returns a specific tap"""
+    serializer_class = TapsListSerializer
+
+
+class TapsList(TapsMixin, generics.ListAPIView):
+    """Returns a list of all of the taps"""
+    serializer_class = TapsListSerializer
```

Comment:
Humm. You're right but I was trying to avoid that.
If we have a single class `TapsView` extending CreateAPIView, RetrieveAPIView, ListAPIView, and GenericAPIView, then how would you write the `URLs` for create, list or get one specific taps? 

If we have the following URLs:
```python
url(r'^/api/v1/taps/(?P<pk>\d+)/?$', api.TapsView.as_view(), name='taps.detail'),
url(r'^/api/v1/taps/?$', api.TapView.as_view(), name='taps.list'),
```
Then, hitting both URLs, we will be able to create a tap and get the list of all taps as well (because TapsView extends CreateAPIView, ListAPIView). I don't think we want to do that do, do we? Maybe you're thinking something else.

Also, `TapsList` uses a different serializer (`TapsListSerializer`) because I want to return the data related to the brewery and price (ie. name, city and no just the PK)

Is there a better way to do that? Maybe you're thinking in something else.